spec/src/main/asciidoc/overview.adoc

```diff
@@ -27,17 +27,18 @@ https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-read
 
 In this scenario health checks are used to determine if a computing node needs to be discarded (terminated, shutdown) and eventually replaced by another (healthy) instance.
 
-The MP-HC architecture consists of a single `/health` endpoint in a MicroProfile runtime that represents the status of
-the entire runtime. This `/health` endpoint is expected to be associated with a configurable context, such as a web
+The MP-HC architecture consists of two `/health/ready` and `/health/live` endpoint in a MicroProfile runtime that respectively represents the readiness and the liveness of the entire runtime.
+For backward compatibility, a 3rd endpoint `/health` is also provided and provide the same information than `/health/live`.
```

Comment:
I think this 3rd route is not just for backward compatibility. This 3rd route is still very important, and I think it justifies keeping the `@Health` annotation.

With appropriate MP Fault Tolerance fallbacks, I can have health checks that fail but my application should still be considered "live" so that it serve requests in a degraded state (in most cases, probably all instances are degraded).

This forces us to require all 3 types of checks: `@Health`, `@Liveness`, and `@Readiness`.

`@Readiness` are checks that keep my app out of rotation when they're failing.

`@Liveness` are checks that kill and restart my app when they're failing.

`@Health` are checks that indicate my app is operating in a degraded manner, but should remain live & ready. (That is, when these checks fail, my app should not be taken out of active rotation and should not be restarted, but my alerting should initiate manual or automated intervention.)

Therefore,
* `/health` should contain only `@Health` checks,
* `/health/ready` (or `/health/readiness`) should contain only `@Readiness` checks, and
* `/health/live` (or `/health/liveness`) should contain only `@Liveness` checks.

I originally thought all existing `@Health` checks were already `@Liveness` checks, but after more careful consideration, I'm pretty sure that that is a wrong assumption.

Reply:
So you basically suggest to reclassify `/health` / `@Health` as custom checks with application-defined / environment-specific meaning? I think that's quite reasonable actually.